apps/bug-cd/src/app/main-navigation.component.ts

```diff
@@ -43,20 +44,34 @@ export class NavigationComponent {
   template: `
     <ng-container *ngIf="info$ | async as info">
       <ng-container *ngIf="info !== null; else noInfo">
-        <app-nav [menus]="getMenu(info)" />
+        <!-- Switching to a variable, instead of calling a function everytime a CD hits -->
+        <app-nav [menus]="menus" />
       </ng-container>
     </ng-container>
 
     <ng-template #noInfo>
-      <app-nav [menus]="getMenu('')" />
+      <!-- Switching to a variable, instead of calling a function everytime a CD hits -->
+      <app-nav [menus]="menus" />
     </ng-template>
   `,
   host: {},
 })
 export class MainNavigationComponent {
   private fakeBackend = inject(FakeServiceService);
-
-  readonly info$ = this.fakeBackend.getInfoFromBackend();
+  info$: any;
+  menus: MenuItem[] = [];
+  constructor() {
+    this.info$ = this.fakeBackend.getInfoFromBackend().pipe(
+      // Using tap operator to update menus via the function every time info$ changes
+      tap((info) => {
+        if (info !== null) {
+          this.menus = this.getMenu(info);
+        } else {
+          this.menus = this.getMenu('');
```

Comment:
You should avoid relying on Angular's change detection and instead purely use an observable.

I think this wouldn't work if `changeDetectionStrategy` was set to `OnPush`. I could be wrong though 😅

Reply:
You coding imperativily, which means you properties are updated at multiple places inside your class. This way of coding feels more natural at first but very hard to debug and maintain.

prefer the declarative approach. meaning declare your properties at one place only. And using observable, those properties will react to any change inside your applications.

In practice:

Don't declare an empty menu, but transform your observable immediatly using the `map` operator to get a list of menuitems. 

Don't declare `info$: any` and assigning the prop later. Assign it immediatly

If you don't understand what I'm saying, you can contact me on linkedin to discuss those point together. I'll be happy to help you.

Reply:
Now updated to declarative approach, thanks!